collect_app/src/androidTest/java/org/odk/collect/android/feature/settings/FormManagementSettingsTest.kt

```diff
@@ -0,0 +1,183 @@
+import org.odk.collect.android.support.pages.FillBlankFormPage
+import org.odk.collect.android.support.pages.FormsDownloadErrorPage
+import org.odk.collect.android.support.pages.MainMenuPage
+import org.odk.collect.android.support.pages.ProjectSettingsPage
+import org.odk.collect.testshared.RecordedIntentsRule
+
+@RunWith(AndroidJUnit4::class)
+class FormManagementSettingsTest {
+    private val testDependencies = TestDependencies()
+    private val notificationDrawer = NotificationDrawerRule()
+    private val rule = CollectTestRule()
+
+    @get:Rule
+    var ruleChain: RuleChain = TestRuleChain.chain(testDependencies)
+        .around(GrantPermissionRule.grant(Manifest.permission.GET_ACCOUNTS))
+        .around(RecordedIntentsRule())
+        .around(notificationDrawer)
+        .around(rule)
+
+    @Test
+    fun whenMatchExactlyEnabled_changingAutomaticUpdateFrequency_changesTaskFrequency() {
+        var deferredTasks = testDependencies.scheduler.deferredTasks
+
+        assertThat(deferredTasks, `is`(empty()))
+
+        val page = MainMenuPage().assertOnPage()
+            .openProjectSettingsDialog()
+            .clickSettings()
+            .clickFormManagement()
+            .clickUpdateForms()
+            .clickOption(R.string.match_exactly)
+
+        deferredTasks = testDependencies.scheduler.deferredTasks
+
+        assertThat(deferredTasks.size, `is`(1))
+
+        val matchExactlyTag = deferredTasks[0].tag
+
+        page.clickAutomaticUpdateFrequency().clickOption(R.string.every_one_hour)
+        deferredTasks = testDependencies.scheduler.deferredTasks
+
+        assertThat(deferredTasks.size, `is`(1))
+        assertThat(deferredTasks[0].tag, `is`(matchExactlyTag))
+        assertThat(deferredTasks[0].repeatPeriod, `is`(1000L * 60 * 60))
+    }
+
+    @Test
+    fun whenPreviouslyDownloadedOnlyEnabled_changingAutomaticUpdateFrequency_changesTaskFrequency() {
+        var deferredTasks = testDependencies.scheduler.deferredTasks
+
+        assertThat(deferredTasks, `is`(empty()))
+
+        val page = MainMenuPage().assertOnPage()
+            .openProjectSettingsDialog()
+            .clickSettings()
+            .clickFormManagement()
+            .clickUpdateForms()
+            .clickOption(R.string.previously_downloaded_only)
+
+        deferredTasks = testDependencies.scheduler.deferredTasks
+
+        assertThat(deferredTasks.size, `is`(1))
+
+        val previouslyDownloadedTag = deferredTasks[0].tag
+        page.clickAutomaticUpdateFrequency().clickOption(R.string.every_one_hour)
+
+        deferredTasks = testDependencies.scheduler.deferredTasks
+
+        assertThat(deferredTasks.size, `is`(1))
+        assertThat(deferredTasks[0].tag, `is`(previouslyDownloadedTag))
+        assertThat(deferredTasks[0].repeatPeriod, `is`(1000L * 60 * 60))
+    }
+
+    @Test
+    fun whenPreviouslyDownloadedOnlyEnabled_checkingAutoDownload_downloadsUpdatedForms_andDisplaysNotification() {
```

Comment:
The auto download tests were here already but looking at it now it feels like they might be better places in `PreviouslyDownloadedOnlyTest`. I'm not sure though. What do you think?

Reply:
Yeah I think it makes sense, done.